detect-nuget-inspector/detect-nuget-inspector/DependencyResolution/Nuget/Resolver/NugetLockFileResolver.cs

```diff
@@ -153,6 +182,30 @@ public DependencyResult Process()
             }
 
             result.Packages = builder.GetPackageList();
+
+            return  FilterUnresolvedDependencies(result);
+        }
+        
+        
+        public DependencyResult FilterUnresolvedDependencies(DependencyResult result)
+        {
+            result.Packages.RemoveAll(packageSet =>
+            {
+                return packageSet.PackageId != null
+                       && directDependenciesMap.ContainsKey(packageSet.PackageId.Name.ToLower())
+                       && !packageSet.PackageId.Version.Equals(directDependenciesMap[packageSet.PackageId.Name.ToLower()]);
```

Comment:
Response to [this comment](https://github.com/blackducksoftware/detect-nuget-inspector/pull/4#discussion_r1284619269): 

directDependenciesMap[<packageID name>] will return a version since it maps packageIDs to versions. 

Replying to[ this comment](https://github.com/blackducksoftware/detect-nuget-inspector/pull/4#discussion_r1284616988):

We do not manipulate the versions when we insert them into the dictionary, so following the same here when doing look ups.